src/cript/api/paginator.py

```diff
@@ -1,58 +1,46 @@
 from json import JSONDecodeError
-from typing import Dict, List, Optional, Union
+from typing import Dict, Union
 from urllib.parse import quote
 
 import requests
 from beartype import beartype
 
-from cript.api.api_config import _API_TIMEOUT
 from cript.api.exceptions import APIError
+from cript.nodes.util import load_nodes_from_json
 
 
 class Paginator:
     """
     Paginator is used to flip through different pages of data that the API returns when searching.
     > Instead of the user manipulating the URL and parameters, this object handles all of that for them.
 
-    When conducting any kind of search the API returns pages of data and each page contains 10 results.
-    This is equivalent to conducting a Google search when Google returns a limited number of links on the first page
-    and all other results are on the next pages.
+    Using the Paginator object, the user can simply and easily flip through the results of the search.
+    The details, that results are listed as pages are hidden from the user.
+    The pages are automatically requested from the API as needed.
 
-    Using the Paginator object, the user can simply and easily flip through the pages of data the API provides.
+    This object implements a python iterator, so `for node in Paginator` works as expected.
+    It will loop through all results of the search, returning the nodes one by one.
 
     !!! Warning "Do not create paginator objects"
         Please note that you are not required or advised to create a paginator object, and instead the
         Python SDK API object will create a paginator for you, return it, and let you simply use it
 
-
-    Attributes
-    ----------
-    current_page_results: List[dict]
-        List of JSON dictionary results returned from the API
-        ```python
-        [{result 1}, {result 2}, {result 3}, ...]
-        ```
     """
 
-    _http_headers: dict
-
-    api_endpoint: str
-
-    # if query or page number are None, then it means that api_endpoint does not allow for whatever that is None
-    # and that is not added to the URL
-    # by default the page_number and query are `None` and they can get filled in
-    query: Union[str, None]
-    _current_page_number: int
-
-    current_page_results: List[dict]
+    _url_path: str
+    _query: str
+    _initial_page_number: Union[int, None]
+    _current_position: int
+    _fetched_nodes: list
+    _number_fetched_pages: int = 0
 
     @beartype
     def __init__(
         self,
-        http_headers: dict,
-        api_endpoint: str,
-        query: Optional[str] = None,
-        current_page_number: int = 0,
+        api,
+        url_path: str,
+        page_number: Union[int, None],
+        query: str,
     ):
```

Comment:
does this object init function take a page number ? just curious in a use case of this instantiation how does a user know what page number to pass in unless there is also a way to get a number of pages?

Reply:
It does take a page number, because some queries do not allow pages to be specified. 
So in that case a user can specify None for the page. 

Also the constructor of the class is mostly hidden from the user as only api.search instantiates them and returns it to the user. 

But just in case, there is default so it's transparent for users.

Reply:
@InnocentBug  could you give a quick example where the call to the paginator would look like to retrieve an already existing node? I left this comment on my sdk update as well because you left an original comment to be implementing api.search or paginator .